pkg/resolver/topology.go

```diff
@@ -0,0 +1,122 @@
+package resolver
+
+import (
+	"fmt"
+
+	"github.com/redhat-appstudio/tssc-cli/pkg/config"
+)
+
+// Topology represents the dependency topology of charts. It is used to determine
+// the order in which charts should be installed.
+type Topology struct {
+	dependencies config.Dependencies // dependency topology
+}
+
+// GetDependencies returns the list of dependencies.
+func (t *Topology) GetDependencies() config.Dependencies {
+	return t.dependencies
+}
+
+// GetDependencyForChart returns the dependency for a given chart name.
+func (t *Topology) GetDependencyForChart(
+	name string,
+) (*config.Dependency, error) {
+	for i := range t.dependencies {
+		if t.dependencies[i].Chart.Name() == name {
+			return &t.dependencies[i], nil
+		}
+	}
+	return nil, fmt.Errorf("dependency not found for chart %s", name)
+}
+
+// exists checks if a dependency exists in the topology.
+func (t *Topology) exists(dependency config.Dependency) bool {
+	for _, d := range t.dependencies {
+		if d.Chart.Name() == dependency.Chart.Name() {
+			return true
+		}
+	}
+	return false
+}
+
+// PrependBefore prepends a list of dependencies before a specific chart.
+func (t *Topology) PrependBefore(name string, dependencies ...config.Dependency) {
```

Comment:
PrependBefore feels reduntant (though in this context it isn't). I would prefer AddBefore or InsertBefore.

Reply:
Well, it merges two concepts: the "prepend" as in the place in the topology, and "before" the desired entry. I think it might sound redundant but expresses the idea in a pragmatic way.